### **Assignment-3**

**Title: Learn Probability Density Functions using Roll-Number-Parameterized Non-Linear Model**

---



In [4]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [27]:
path='/content/drive/MyDrive/predictive-analysis/data.csv'

In [28]:
import numpy as np
import pandas as pd

In [29]:
pd.read_csv(path,encoding='latin1')

/tmp/ipython-input-1879989015.py:1: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  pd.read_csv(path,encoding='latin1')


stn_code       sampling_date                        state   location  \
0         150.0  February - M021990               Andhra Pradesh  Hyderabad   
1         151.0  February - M021990               Andhra Pradesh  Hyderabad   
2         152.0  February - M021990               Andhra Pradesh  Hyderabad   
3         150.0     March - M031990               Andhra Pradesh  Hyderabad   
4         151.0     March - M031990               Andhra Pradesh  Hyderabad   
...         ...                 ...                          ...        ...   
435737     SAMP            24-12-15                  West Bengal   ULUBERIA   
435738     SAMP            29-12-15                  West Bengal   ULUBERIA   
435739      NaN                 NaN  andaman-and-nicobar-islands        NaN   
435740      NaN                 NaN                  Lakshadweep        NaN   
435741      NaN                 NaN                      Tripura        NaN   

                                           agency  \
0                                             NaN   
1                                             NaN   
2                                             NaN   
3                                             NaN   
4                                             NaN   
...                                           ...   
435737  West Bengal State Pollution Control Board   
435738  West Bengal State Pollution Control Board   
435739                                        NaN   
435740                                        NaN   
435741                                        NaN   

                                      type   so2   no2   rspm  spm  \
0       Residential, Rural and other Areas   4.8  17.4    NaN  NaN   
1                          Industrial Area   3.1   7.0    NaN  NaN   
2       Residential, Rural and other Areas   6.2  28.5    NaN  NaN   
3       Residential, Rural and other Areas   6.3  14.7    NaN  NaN   
4                          Industrial Area   4.7   7.5    NaN  NaN   
...                                    ...   ...   ...    ...  ...   
435737                               RIRUO  22.0  50.0  143.0  NaN   
435738                               RIRUO  20.0  46.0  171.0  NaN   
435739                                 NaN   NaN   NaN    NaN  NaN   
435740                                 NaN   NaN   NaN    NaN  NaN   
435741                                 NaN   NaN   NaN    NaN  NaN   

              location_monitoring_station  pm2_5        date  
0                                     NaN    NaN  1990-02-01  
1                                     NaN    NaN  1990-02-01  
2                                     NaN    NaN  1990-02-01  
3                                     NaN    NaN  1990-03-01  
4                                     NaN    NaN  1990-03-01  
...                                   ...    ...         ...  
435737  Inside Rampal Industries,ULUBERIA    NaN  2015-12-24  
435738  Inside Rampal Industries,ULUBERIA    NaN  2015-12-29  
435739                                NaN    NaN         NaN  
435740                                NaN    NaN         NaN  
435741                                NaN    NaN         NaN  

[435742 rows x 13 columns]

In [30]:
data=pd.read_csv(path,encoding='latin1')['no2']

/tmp/ipython-input-3421846029.py:1: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  data=pd.read_csv(path,encoding='latin1')['no2']


In [31]:
print(data)

0         17.4
1          7.0
2         28.5
3         14.7
4          7.5
          ... 
435737    50.0
435738    46.0
435739     NaN
435740     NaN
435741     NaN
Name: no2, Length: 435742, dtype: float64


In [32]:
print(data.size)
print(data.isna().sum())

435742
16233


this shows that out of 435742 values 16233 are NaN values.

In [26]:
x=data.dropna()

In [34]:
print(x.size)

419509


In [38]:
#z = Tr(x) = x + arsin(brx)
r=102317096
a_r=0.05*(r%7)
b_r=0.3*(r%5+1)

print(a_r)
print(b_r)

0.0
0.6


In [40]:
z=x+a_r*(np.sin(b_r*x))
print(z)

0         17.4
1          7.0
2         28.5
3         14.7
4          7.5
          ... 
435734    44.0
435735    44.0
435736    45.0
435737    50.0
435738    46.0
Name: no2, Length: 419509, dtype: float64


In [42]:
mu=np.mean(z)
variance=np.var(z)
lambda_val=1/(2*variance)
c=np.sqrt(lambda_val/np.pi)

print(mu)
print(variance)
print(c)

25.809622897811263
342.36339017380317
0.02156087623931353
